File: native-modules/zalopay-sdk/ios/ZaloPayModule.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(ZaloPaySDK, RCTEventEmitter)

RCT_EXTERN_METHOD(initWithAppId:(NSString *)appId
                  uriScheme:(NSString *)uriScheme
                  environment:(NSString *)environment
                  callback:(RCTResponseSenderBlock)callback)

RCT_EXTERN_METHOD(payOrder:(NSString *)zpTransToken
                  callback:(RCTResponseSenderBlock)callback)

@end

File: native-modules/zalopay-sdk/ios/ZaloPayModule.swift
import Foundation
import ZaloPaySDK

@objc(ZaloPaySDK)
class ZaloPaySDK: RCTEventEmitter {
    override func supportedEvents() -> [String]! {
        return ["ZaloPayEvent"]
    }

    @objc
    func initWithAppId(_ appId: String, uriScheme: String, environment: String, callback: RCTResponseSenderBlock) {
        ZaloPaySDK.shared().init(withAppId: appId, uriScheme: uriScheme, environment: environment == "SANDBOX" ? .sandbox : .production)
        callback([nil, "Initialized"])
    }

    @objc
    func payOrder(_ zpTransToken: String, callback: RCTResponseSenderBlock) {
        ZaloPaySDK.shared().payOrder(zpTransToken) { errorCode, message in
            callback([nil, NSNumber(value: errorCode), message ?? ""])
            self.sendEvent(withName: "ZaloPayEvent", body: [
                "errorCode": errorCode,
                "message": message ?? ""
            ])
        }
    }
}